app/src/main/java/de/tum/in/tumcampusapp/managers/CafeteriaMenuManager.java

```diff
@@ -219,4 +212,63 @@ private void replaceIntoDb(CafeteriaMenu c) {
                         Utils.getDateString(c.date), c.typeShort, c.typeLong,
                         String.valueOf(c.typeNr), c.name});
     }
+
+    /**
+     * Prepares a bundle, which can be sent to the FavoriteDishAlarmScheduler, which contains all necessary
+     * information to schedule the FavoriteDishAlarms. Its procedure is the following: Get the names
+     * of all the favorite dishes and their corresponding mensaId (the user flags a food as favorite,
+     * which also stores the mensaId). By assuming that the user will only rate the food as a favorite,
+     * if he actually goes to that specific mensa. The alarm is then stored and scheduled, if it's not
+     * scheduled already.
+     *
+     * @param completeReschedule
+     * True if all currently scheduled alarms should be discarded, False if not
+     */
+    public void scheduleFoodAlarms(boolean completeReschedule){
+        FavoriteFoodAlarmStorage favoriteFoodAlarmStorage = FavoriteFoodAlarmStorage.getInstance(mContext);
+        if(completeReschedule){
+            favoriteFoodAlarmStorage.cancelOutstandingAlarms();
+        }
+        Cursor favoriteFoodWhere = db.rawQuery("SELECT mensaId,dishName FROM favorite_dishes GROUP BY mensaId,dishName",null);
+        while (favoriteFoodWhere.moveToNext()){
+            int mensaId = favoriteFoodWhere.getInt(0);
+            String dishName = favoriteFoodWhere.getString(1);
+            Cursor upcomingServings = db.rawQuery("SELECT date FROM cafeterias_menus WHERE date >= date('now','localtime') AND mensaId = ? AND name = ?",new String[]{""+mensaId, dishName});
```

Comment:
This should probably be a join instead of queries in a loop over the result of a query...